### PR TITLE
P20-706: Removes 'All' page, adds more page sizes, and adds CSV generation to permissions query page

### PR DESCRIPTION
**Pull request by wilkie**
Reported first in Slack: https://codedotorg.slack.com/archives/C04540KNGDQ/p1707240207180699

Essentially, we had a "All" button for the pagination... which failed due to the logic trying to use "All" as a number. However, it became clear that what they really needed was a larger pagination beyond 500. And they were working around it by manually forcing a page size of that magnitude.

Ultimately they want the data to create a CSV. Therefore, I asked if we could just... provide a download button.

This does the following:

* Removes the ill-advised "All" page option
* Adds page sizes in the thousands
* Adds a CSV download that will download a CSV representing the table on the current page

## Links

- jira ticket: [P20-706](https://codedotorg.atlassian.net/browse/P20-706)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/views/admin_users/permissions_form.html.haml [hunk 1]
```diff
@@ -55,7 +55,8 @@
   %button.btn{type: 'submit'}
     %i.fa.fa-search
 -if @users_with_permission
-  = page_header 'users', @users_with_permission, permitted_params: [:permission]
+  - download_csv = link_to('Page as CSV', permissions_csv_path(params.permit([:permission, :page_size, :page, :search_term])), class: 'btn btn-default')
```
Comment (carl-codeorg) [inline dashboard/app/views/admin_users/permissions_form.html.haml:58]: This button text might be more intuitive as "Download CSV"  or "Download Page as CSV".

**Review by carl-codeorg — APPROVED** [review]
LGTM, just one minor suggestion for the button text.

File: dashboard/app/views/admin_users/permissions_form.html.haml (hunk 1 above)
Comment (wilkie) [reply dashboard/app/views/admin_users/permissions_form.html.haml:58]: Yeah, I appreciate this comment... I had "Download Page as CSV" at one point and it felt comically long and would get in the way of adding any buttons to that header in the future. And just "Download CSV" might be surprising that the CSV is just the subset on the screen. It's not a user-facing button, so after thinking it through, I think I'll keep it as "Page as CSV" and not overthink it too much.

Comment (wilkie) [reply dashboard/app/views/admin_users/permissions_form.html.haml:58]: Plus I'm a tad lazy to think too much about a feature for an internal tool. :)

**Pull request merged**